dlt/common/destination/reference.py

```diff
@@ -453,9 +450,9 @@ def capabilities(self) -> DestinationCapabilitiesContext:
         ...
 
     @property
-    def name(self) -> str:
+    def destination_name(self) -> str:
         """The destination name will either be explicitely set while creating the destination or will be taken from the type"""
-        return self.config_params.get("name") or self.destination_type
+        return self.config_params.get("destination_name") or self.destination_type
```

Comment:
use ` def to_name(ref: TDestinationReferenceArg) -> str:` to get name from type. use it everywhere you do that